File: app/src/main/java/com/prey/actions/alert/AlertConfig.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2019 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.alert;

import android.content.Context;

public class AlertConfig {

    private static AlertConfig cachedInstance = null;
    private Context ctx;
    private int notificationId = 100;

    private AlertConfig(Context ctx) {
        this.ctx = ctx;
    }

    public static synchronized AlertConfig getAlertConfig(Context ctx) {
        if (cachedInstance == null) {
            synchronized (AlertConfig.class) {
                if (cachedInstance == null)
                    cachedInstance = new AlertConfig(ctx);
            }
        }
        return cachedInstance;
    }

    public int getNotificationId() {
        notificationId = notificationId + 1;
        return notificationId;
    }

}

File: app/src/main/java/com/prey/actions/alert/kotlin/AlertConfig.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.alert.kotlin

import android.content.Context

class AlertConfig(private val context: Context) {
    private var nextNotificationId = 100

    fun getNextNotificationId(): Int {
        return nextNotificationId++
    }

    companion object {
        private var instance: AlertConfig? = null

        @Synchronized
        fun getInstance(context: Context): AlertConfig {
            if (instance == null) {
                instance = AlertConfig(context)
            }
            return instance!!
        }
    }
}